### PR TITLE
feat: Add setting to auto approve PRs in Azure DevOps

### LINKED ISSUE #9685: Implement auto approve PRs in Azure DevOps
**What would you like Renovate to be able to do?**

When Renovate creates a PR, we would like it to be auto-completed as soon as the build succeeds, bypassing our mandatory reviewer policies. When the branch policies of a repository require a PR to be approved before completing, the azureAutoComplete setting is not enough by itself.

**Did you already have any implementation ideas?**

Renovate could automatically approve PRs with a setting. E.g. azureAutoApprove, that will approve the PR with the access token Renovate is using. This follows the example of azureAutoComplete configuration that is already available.

We are in the process of creating a PR for this based on the above implementation.

### PR DESCRIPTION
**Pull request by TiagoPT**
## Changes:

This PR introduces a new Azure DevOps specific setting that allows the PR to be automatically approved, to bypass the branch policies. 
See Issue #9685 for more details.

## Context:

Closes #9685 

## Documentation (please check one with an [x])

- [X] I have updated the documentation

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
You need to fix your commit author and commiter values

**Comment by TiagoPT** [comment]
Fixed the commit author details.

File: lib/platform/azure/index.ts [hunk 1]
```diff
@@ -416,6 +416,19 @@ export async function createPr({
       pr.pullRequestId
     );
   }
+  if (platformOptions?.azureAutoApprove) {
+    pr = await azureApiGit.createPullRequestReviewer(
+      {
+        reviewerUrl: pr.createdBy.url,
+        vote: 10,
```
Comment (JamieMagee) [inline lib/platform/azure/index.ts:423]: tiny nit: this is a magic number. I know it comes from Azure DevOps, but can you either leave a comment where you found it ([here?](https://docs.microsoft.com/en-us/rest/api/azure/devops/git/pull%20request%20reviewers/create%20pull%20request%20reviewer?view=azure-devops-rest-6.0)) or make this a constant with a good name.

**Review by JamieMagee — DISMISSED** [review]

File: lib/platform/gitlab/index.spec.ts [hunk 2]
```diff
@@ -1289,6 +1289,7 @@ describe(getName(__filename), () => {
         prBody: 'the-body',
         labels: [],
         platformOptions: {
+          azureAutoApprove: false,
```
Comment (viceice) [inline lib/platform/gitlab/index.spec.ts:1292]: Revert this

**Review by viceice — CHANGES_REQUESTED** [review]
Needs test fix

**Review by HonkingGoose — DISMISSED** [review]
Docs looks good to me.

File: lib/platform/azure/index.ts (hunk 1 above)
Comment (TiagoPT) [reply lib/platform/azure/index.ts]: Good point @JamieMagee. I used the source you linked for that magic number, and fixed it by adding AzurePrVote enum to make it clear what it means.

File: lib/platform/gitlab/index.spec.ts (hunk 2 above)
Comment (TiagoPT) [reply lib/platform/gitlab/index.spec.ts]: Reverted.

File: lib/platform/azure/index.ts [hunk 3]
```diff
@@ -35,6 +35,7 @@ import { smartTruncate } from '../utils/pr-body';
 import * as azureApi from './azure-got-wrapper';
 import * as azureHelper from './azure-helper';
 import type { AzurePr } from './types';
+import { AzurePrVote } from './types';
```
Comment (viceice) [inline lib/platform/azure/index.ts:38]: Merge with lone before and remove `type`

**Review by viceice — COMMENTED** [review]
A small thing

File: lib/platform/azure/index.ts (hunk 3 above)
Comment (TiagoPT) [reply lib/platform/azure/index.ts]: I have merged the type import statements.

**Review by viceice — APPROVED** [review]

**Comment by TiagoPT** [comment]
Hey everyone, can I have an update on this PR? I am eager to use this feature on my repositories.

**Pull request merged**